src/scanpy/plotting/_tools/__init__.py

```diff
@@ -474,15 +489,21 @@ def rank_genes_groups(
         if count % n_panels_x == 0:
             ax.set_ylabel("score")
 
-    if sharey is True:
+    if sharey is True and ax is not None:
```

Comment:
What case is `ax` `None` here? It looks like `ax` is set inside the above loop so that would mean `group_names` is a length-0 object?

Reply:
That could happen if the degenerate case `group_names==[]`, but I didn’t actually try what happens then